neqo-transport/src/recovery.rs

```diff
@@ -412,7 +412,7 @@ impl LossRecoverySpace {
             .sent_packets
             .iter_mut()
             // BTreeMap iterates in order of ascending PN
-            .take_while(|(&k, _)| Some(k) < largest_acked)
+            .take_while(|(&k, _)| largest_acked.is_none() || Some(k) < largest_acked)
```

Comment:
This sort of change really needs to be in a separate patch so that we can look at the recovery tests in more detail.

Reply:
Yeah, I realized this is getting big. I will break it into individual chunks.